vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLUnixDomainSocketTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.mysqlclient;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.sqlclient.PoolOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.shaded.org.apache.commons.lang.SystemUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assume.assumeTrue;
+
+@RunWith(VertxUnitRunner.class)
+public class MySQLUnixDomainSocketTest {
+
+  private static final String unixSocketDirectory = System.getProperty("unix.socket.directory");
+
+  private MySQLPool client;
+  private MySQLConnectOptions options;
+
+  @Before
+  public void before() {
+    assumeTrue(SystemUtils.IS_OS_UNIX);
+    options = new MySQLConnectOptions()
+      .setUser("root")
+      .setPassword("password");
+    if (unixSocketDirectory != null && !unixSocketDirectory.isEmpty()) {
+      options.addProperty("socket", unixSocketDirectory);
+    }
+    assumeTrue(options.isUsingDomainSocket());
+  }
+
+  @After
+  public void after() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  @Test
+  public void uriTest(TestContext context) throws UnsupportedEncodingException {
+    String path = URLEncoder.encode(options.getProperties().get("socket"), "UTF-8");
+    String uri = "mysql://root:password@localhost?socket=" + path;
+    client = MySQLPool.pool(uri);
+    client.getConnection(context.asyncAssertSuccess(pgConnection -> pgConnection.close()));
```

Comment:
`pgConnection` should be renamed in this class

Reply:
Done